packages/@lwc/compiler/src/transformers/javascript.ts

```diff
@@ -85,8 +103,25 @@ export default function scriptTransform(
             // an error when the generated code is over 500KB.
             compact: false,
             plugins,
+            parserOpts: {
+                ...(experimentalErrorRecoveryMode ? { errorRecovery: true } : {}),
+            },
         })!;
+
+        const lwcErrors = extractLwcErrors(result);
+
+        if (!experimentalErrorRecoveryMode || lwcErrors.length === 0) {
+            return {
```

Comment:
nit: I personally prefer having the return statement at the end even if it means `result` isn't a `const`. It forces a cleaner separation between the 'successful' codepath and the 'error' code paths, and doesn't hide the main successful return value in the middle.